sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -284,6 +284,18 @@ def fetch_20newsgroups(data_home=None, subset='train', categories=None,
     if 'quotes' in remove:
         data.data = [strip_newsgroup_quoting(text) for text in data.data]
 
+    if len(remove) > 0:
```

Comment:
Would it be possible to make this selection in the previous filtering? Something along these lines:

```python
    strip_funcs = {
        "headers": strip_newsgroup_header,
        "footers": strip_newsgroup_footer,
        "quotes": strip_newsgroup_quoting,
    }

    for strip_type in remove:
        tmp_data = []
        for text in data:
            stripped_text = strip_funcs[strip_type](text)
            if not stripped_text:
                tmp_data.append(stripped_text)
        data.data = tmp_data
```